Src/DependencyCollector/Shared/Implementation/HttpParsers/AzureSearchHttpParser.cs

```diff
@@ -0,0 +1,203 @@
+            ["GET /datasources"] = "List data sources",
+            ["GET /indexers"] = "List indexers",
+            ["POST /indexers/*/reset"] = "Reset indexer",
+            ["POST /indexers/*/run"] = "Run indexer",
+            ["PUT /datasources/*"] = "Update data source",
+            ["PUT /indexers/*"] = "Update indexer",
+
+            // Service operations
+            ["GET /servicestats"] = "Get service statistics",
+
+            // Skillset operations
+            ["POST /skillsets/*"] = "Create skillset",
+            ["DELETE /skillsets/*"] = "Delete skillset",
+            ["GET /skillsets/*"] = "Get skillset",
+            ["GET /skillsets"] = "List skillsets",
+            ["PUT /skillsets/*"] = "Update skillset",
+
+            // Synonym operations
+            ["POST /synonymmaps"] = "Create synonym map",
+            ["PUT /synonymmaps/*"] = "Update synonym map",
+            ["GET /synonymmaps"] = "List synonym maps",
+            ["GET /synonymmaps/*"] = "Get synonym map",
+            ["DELETE /synonymmaps/*"] = "Delete synonym map"
+        };
+
+        /// <summary>
+        /// Tries parsing given dependency telemetry item. 
+        /// </summary>
+        /// <param name="httpDependency">Dependency item to parse. It is expected to be of HTTP type.</param>
+        /// <returns><code>true</code> if successfully parsed dependency.</returns>
+        internal static bool TryParse(ref DependencyTelemetry httpDependency)
+        {
+            var name = httpDependency.Name;
+            var host = httpDependency.Target;
+            var url = httpDependency.Data;
+
+            if (name == null || host == null || url == null)
+            {
+                return false;
+            }
+
+            if (!HttpParsingHelper.EndsWithAny(host, AzureSearchHostSuffixes))
+            {
+                return false;
+            }
+
+            ////
+            //// Azure Search REST API: https://docs.microsoft.com/en-us/rest/api/searchservice/
+            ////
+
+            HttpParsingHelper.ExtractVerb(name, out var verb, out var nameWithoutVerb, AzureSearchSupportedVerbs);
+
+            var resourcePath = ParseResourcePath(nameWithoutVerb);
+
+            // populate properties
+            foreach (var resource in resourcePath)
+            {
+                if (resource.Value != null)
+                {
+                    var propertyName = GetPropertyNameForResource(resource.Key);
+                    if (propertyName != null)
+                    {
+                        httpDependency.Properties[propertyName] = resource.Value;
+                    }
+                }
+            }
+
+            var operation = BuildOperationMoniker(verb, resourcePath);
+            var operationName = GetOperationName(operation);
+
+            httpDependency.Type = RemoteDependencyConstants.AzureSearch;
+            httpDependency.Name = string.IsNullOrEmpty(operationName) ? httpDependency.Target : operationName;
```

Comment:
```suggestion
            httpDependency.Name = string.IsNullOrEmpty(operationName) ? name : operationName;
```

If operationName is not parsed, let's keep dependency name unchanged